### PR TITLE
Scidavis

### PR DESCRIPTION
**Pull request by rardiol**
###### Things done
- [x] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Python isn't working yet, it just SIGSEGVs as soon as python is selected as scripting language

### CODE REVIEW

File: pkgs/applications/science/math/scidavis/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{stdenv, fetchurl, qwt, gsl, qwtPlot3D, muparser, zlib, qt4, qmake4Hook, python2, python2Packages
```
Comment (FRidh) [inline pkgs/applications/science/math/scidavis/default.nix]: please don't pass in python2 as python2Packages already has the attribute python

Comment (rardiol) [inline pkgs/applications/science/math/scidavis/default.nix]: fixed

**Comment by Mic92** [comment]
is this finished already?

**Comment by rardiol** [comment]
If by "finished" you mean the python problem being fixed then no.

**Comment by Mic92** [comment]
do you think this application would be useful without python support?

File: pkgs/applications/science/math/scidavis/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  qmakeFlags = "CONFIG+=python";
+
+  installFlags = "INSTALL_ROOT=$(out)";
+
+  postInstall = "mv $out/usr/* $out";
+
+  buildInputs = [
+     qwt gsl muparser qt4 zlib qmake4Hook qwtPlot3D
+     python2Packages.python python2Packages.sip python2Packages.pyqt4 #Support for python
```
Comment (FRidh) [inline pkgs/applications/science/math/scidavis/default.nix]: I suggest you use `python.withPackages` to create a Python env where all modules can be found

**Comment by rardiol** [comment]
Yes, the default scripting language is working, so it's useful.

File: pkgs/applications/science/math/scidavis/default.nix (hunk 2 above)
Comment (rardiol) [inline pkgs/applications/science/math/scidavis/default.nix]: I copied from gnome-tweak-tool and now it SIGSEGV at startup:

```
{stdenv, fetchurl, qwt, gsl, qwtPlot3D, muparser, zlib, qt4, qmake4Hook, python2Packages
}:

let
  python = python2Packages.python.withPackages ( ps: with ps; [ sip pyqt4 ]); #Support for python
in stdenv.mkDerivation rec {
  version = "1.14";
  name = "scidavis-${version}";

  src = fetchurl {
    url = "mirror://sourceforge/project/scidavis/SciDAVis/${version}/scidavis-${version}.tar.gz";
    sha256 = "0904dfz0ijgqvx216wy7qic2hhnm0inkmwi9484d1517sb0bdz55";
  };

  qmakeFlags = "CONFIG+=python";

  installFlags = "INSTALL_ROOT=$(out)";

  postInstall = "mv $out/usr/* $out";

  buildInputs = [
     qwt gsl muparser qt4 zlib qmake4Hook qwtPlot3D
     python
  ];

  meta = with stdenv.lib; {
    homepage = http://scidavis.sourceforge.net/;
    description = "A free application for Scientific Data Analysis and Visualization";
    license = licenses.gpl2Plus;
    platforms = platforms.all;
  };
}

```

**Comment by Mic92** [comment]
do you want to disable python support if its not working?

**Comment by rardiol** [comment]
I wanted a fairy to come and fix it for me, but looks like it's not happening. So yes, python support removed.

File: pkgs/applications/science/math/scidavis/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+#  qmakeFlags = "CONFIG+=python";
+
+  installFlags = "INSTALL_ROOT=$(out)";
+
+  postInstall = "mv $out/usr/* $out";
+
+  buildInputs = [
+     qwt gsl muparser qt4 zlib qmake4Hook qwtPlot3D
+ #    python2Packages.python python2Packages.sip python2Packages.pyqt4 #Support for python
```
Comment (FRidh) [inline pkgs/applications/science/math/scidavis/default.nix]: use `python.withPackages( ps; with ps; [ pyqt4 sip ])`

Comment (FRidh) [inline pkgs/applications/science/math/scidavis/default.nix]: likely that's enough since any shebangs pointing to python will then use environment

**Review by FRidh — APPROVED** [review]

**Comment by FRidh** [comment]
This has been open for quite a while. @rardiol, please open a new PR if you want to get this in.

**Pull request closed without merging**